### ISSUE TITLE
[Bug]: Xiaomi Wireless Switch Key "XMWXKG01LM"  not change status "Unknown"

### ISSUE THREAD

**Issue by Kazarnikov** [custom-components/ble_monitor#1268]
### What happened?

The button does not change the status when pressed, it is always "Unknown".
It is impossible to use actions in automations.

Additionally added a log line:
2023-12-07 22:35:51.204 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] payload: b'\rN\x00', data: b'\x18\x16\x95\xfeXY\x87#\xe9\ri*<\xc2\x18\x1e\x13\xc1\x08\x00\x00\xd2\xa2I\x81', i: 15, hex: 181695fe58598723e90d692a3cc2181e13c1080000d2a24981, device_id: 9095, device_type: XMWXKG01LM

yaml:
          mac: '18:C2:3C:2A:69:0D'
          name: 'button_switch_one'
          encryption_key: 'c533a5ab361b0a24de4d21d1d9a3d8a1'

Current version: 2.0.0_0014
Region: Chine
Operating modes: Standard mode

The clicks are recorded in the log in order:
 - single
 - double
 - triple
 - quadruple
 - hold

### Sensor type

Xiaomi Wireless Switch Key (Bluetooth) XMWXKG01LM

### Relevant log output

```shell
2023-12-07 23:01:35.625 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:36.125 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:38.175 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:39.375 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] payload: b'\x0cN\x00', data: b'\x18\x16\x95\xfeXY\x87#\xff\ri*<\xc2\x18v\xd7\xa7\x08\x00\x00`$\xe7W', i: 15, hex: 181695fe58598723ff0d692a3cc21876d7a70800006024e757, device_id: 9095, device_type: XMWXKG01LM
2023-12-07 23:01:39.376 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -70, 'mac': '18C23C2A690D', 'type': 'XMWXKG01LM', 'packet': 255, 'firmware': 'Xiaomi (MiBeacon V5 encrypted)', 'data': True}
2023-12-07 23:01:41.475 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:41.975 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:42.475 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] payload: b'\rN\x00', data: b'\x18\x16\x95\xfeXY\x87#\x01\ri*<\xc2\x18\xf3\x97\xdd\t\x00\x00y\x82k\x9d', i: 15, hex: 181695fe58598723010d692a3cc218f397dd09000079826b9d, device_id: 9095, device_type: XMWXKG01LM
2023-12-07 23:01:42.478 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -68, 'mac': '18C23C2A690D', 'type': 'XMWXKG01LM', 'packet': 1, 'firmware': 'Xiaomi (MiBeacon V5 encrypted)', 'data': True}
2023-12-07 23:01:44.525 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:44.775 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:46.276 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'rssi': -66, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
2023-12-07 23:01:46.276 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -66, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
2023-12-07 23:01:47.325 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:47.575 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] Unknown Xiaomi device found. Data: 111695fe3058430b0120336e452870280100
2023-12-07 23:01:50.325 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'rssi': -74, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
2023-12-07 23:01:50.326 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -74, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
2023-12-07 23:01:52.298 DEBUG (Thread-3) [custom_components.ble_monitor.ble_parser.xiaomi] payload: b'\x0eN\x00', data: b'\x18\x16\x95\xfeXY\x87#\x03\ri*<\xc2\x18%\x88$\t\x00\x00\xa3`\xb8\xa1', i: 15, hex: 181695fe58598723030d692a3cc218258824090000a360b8a1, device_id: 9095, device_type: XMWXKG01LM
2023-12-07 23:01:52.301 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -76, 'mac': '18C23C2A690D', 'type': 'XMWXKG01LM', 'packet': 3, 'firmware': 'Xiaomi (MiBeacon V5 encrypted)', 'data': True}
2023-12-07 23:01:56.527 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Data binary sensor received: {'rssi': -66, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
2023-12-07 23:01:56.527 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Data measuring sensor received: {'rssi': -66, 'packet': 'no packet id', 'temperature': 26.5, 'humidity': 44.7, 'battery': 42, 'mac': '582D34525F34', 'type': 'CGD1', 'firmware': 'Qingping', 'data': True}
```

**Comment by Ernst79** [comment]
Hmmm, I can see that it returns a MiBeacon object `0e4e00` (after decryption). 
`0e4e` = long click
`00` = length of the data (normally), but as it is zero, it won't continue in the code. 

Similar for `0d4e00` (Double click) and  `0c4e00` (Single click)

I will fix this

**Comment by Ernst79** [comment]
Fixed in 12.6.5

**Comment by Kazarnikov** [comment]
Working. Thanks
